dm/loader/lightning.go

```diff
@@ -202,11 +205,37 @@ func (l *LightningLoader) runLightning(ctx context.Context, cfg *lcfg.Config) er
 	}
 
 	var opts []lightning.Option
+	if l.cfg.MetricsFactory != nil {
+		opts = append(opts,
+			lightning.WithPromFactory(
+				promutil.NewWrappingFactory(
+					l.cfg.MetricsFactory,
+					"",
+					prometheus.Labels{"task": l.cfg.Name, "source_id": l.cfg.SourceID},
+				)),
+			lightning.WithPromRegistry(promutil2.NewNoopRegistry()))
```

Comment:
why use noop registry?

Reply:
If dataflow engine provides a Factory, the Factory will handle register and deregister, no other registerer is needed.

Reply:
maybe add a comment

Reply:
https://github.com/pingcap/tiflow/pull/5869/commits/75bd932c9e2ed00c04283c62432a45cbb5ea577b